nemo/collections/asr/data/audio_to_text_lhotse_prompted.py

```diff
@@ -25,6 +26,18 @@
 from nemo.collections.common.tokenizers.canary_tokenizer import CANARY_SPECIAL_TOKENIZER
 
 
+@dataclass
+class PromptedAudioToTextMiniBatch:
```

Comment:
Per @titu1994 from offthread convo.

```


PTL will not auto cast tensors tht are inside data classes, only inside simpler structures like lists, tuples and dicts.
We can try to use TypedDict collection, that might work```


[11:12 AM](https://nvidia.slack.com/archives/D02KMNFH6UC/p1721844762008059)
Just use torch.Tensor or Any as types if needed
11:13 AM
We need to test if PTL can handle the change properly and still continue to cast inputs automatically for the batch
[11:13 AM](https://nvidia.slack.com/archives/D02KMNFH6UC/p1721844800900519)
Worst comes to worse we can ask PTL to check for and cast tensors to service from inside TypedDicts though that will take time

'''

Reply:
It worked fine when I ran this, maybe they added dataclass support later.

Reply:
When you were running were you checking casting at all?

Reply:
I launched a multi-GPU training with this change and it runs fine in terms of GPU utilization, so the batch had to be moved to CUDA.

Reply:
Marvelous